### PR TITLE
SPARKC-536 UDT converters optimization

### PR DESCRIPTION
**Pull request by glutamatt**
### [SPARKC-536](https://datastax-oss.atlassian.net/browse/SPARKC-536)

Avoid overhead of repeated `columnTypes(i).converterToCassandra` which lead to synchronized reflection costly methods.

Profiling a spark job wich prepare a sstable from dataframe with many udt into collections, with project https://github.com/jparkie/Spark2Cassandra , up to 90% of the time is spent into `converterToCassandra` for the udt columns conversion

I just expand the memoization of converters we already have in `DriverUDTValueConverter` to the case class UDTFieldDef converter generator in converterToCassandra

I wasn't able to register on the jira project (in order to create a issue to associate to the PR) 
> datastax-oss.atlassian.net has reached its license limit.

[Build Status](https://travis-ci.org/glutamatt/spark-cassandra-connector)

### CODE REVIEW

**Comment by glutamatt** [comment]
CPU sample **before** the optimization

image

**After** => **72 % faster** on the writeRowsToSSTables method

image

File: spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/UserDefinedType.scala [hunk 1]
```diff
@@ -33,14 +33,16 @@ case class UserDefinedType(name: String, columns: IndexedSeq[UDTFieldDef])
   def scalaTypeTag = SparkReflectionLock.synchronized { implicitly[TypeTag[UDTValue]] }
   def cqlTypeName = name
 
+  val fieldConverters = columnTypes.map(_.converterToCassandra)
+
```
Comment (abenicho) [inline spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/UserDefinedType.scala]: remove empty line

**Comment by RussellSpitzer** [comment]
I think this is a great fix, could you please make a corresponding Jira? We don't actively check github pullrequests which is why no one from DS has looked at this.

**Comment by ManuelWeiss** [comment]
@RussellSpitzer That's a big mistake. Can you also please get https://github.com/datastax/spark-cassandra-connector/pull/1171 merged?

**Comment by glutamatt** [comment]
> could you please make a corresponding Jira?

Done -> [SPARKC-536](https://datastax-oss.atlassian.net/browse/SPARKC-536)

**Comment by RussellSpitzer** [comment]
Excellent! Thank you so much

**Comment by bcantoni** [comment]
@glutamatt thanks for your note about the https://datastax-oss.atlassian.net JIRA instance being at its limit -- we've cleaned out some inactive users to make room for new accounts if you'd like to create one now.

**Pull request merged**